File: JavaScript/TheHurdleRace.js
const hurdleRace = (k, height) => {
    let potions = 0;

    height.sort((a,b)=> a - b);

    if(height[height.length-1] >= k) potions = height[height.length-1] - k;
    
    return potions;
}

function testHurdleRace() {

    const testCases = [
        { k: 4, height: [1, 6, 3, 5, 2], expected: 2 },
        { k: 7, height: [2, 5, 4, 8, 9], expected: 2 },
        { k: 3, height: [4, 2, 7, 6, 5], expected: 4 },
        { k: 10, height: [11, 12, 9, 8, 15], expected: 5 },
        { k: 5, height: [7, 5, 5, 6, 8], expected: 3 }
    ];

    for (const testCase of testCases) {
        const result = hurdleRace(testCase.k, testCase.height);
        const pass = result === testCase.expected;
        console.log(`Test Case: k=${testCase.k}, height=[${testCase.height}], Expected: ${testCase.expected}, Result: ${result}, Pass: ${pass ? 'Yes' : 'No'}`);
    }
}

testHurdleRace();

File: Java/TheHurdleRace.java
import java.util.Arrays;
import java.util.List;

public class TheHurdleRace {

    public static int hurdleRace(int k, List<Integer> height) {
        int potions = 0;

        height.sort(Integer::compareTo);

        if (height.get(height.size() - 1) >= k) {
            potions = height.get(height.size() - 1) - k;
        }

        return potions;
    }

    public static void testHurdleRace() {
        List<TestCase> testCases = Arrays.asList(
                new TestCase(4, Arrays.asList(1, 6, 3, 5, 2), 2),
                new TestCase(7, Arrays.asList(2, 5, 4, 8, 9), 2),
                new TestCase(3, Arrays.asList(4, 2, 7, 6, 5), 4),
                new TestCase(10, Arrays.asList(11, 12, 9, 8, 15), 5),
                new TestCase(5, Arrays.asList(7, 5, 5, 6, 8), 3));

        for (TestCase testCase : testCases) {
            int result = hurdleRace(testCase.k, testCase.height);
            boolean pass = result == testCase.expected;
            System.out.println("Test Case: k=" + testCase.k + ", height=" + testCase.height +
                    ", Expected: " + testCase.expected + ", Result: " + result +
                    ", Pass: " + (pass ? "Yes" : "No"));
        }
    }

    public static void main(String[] args) {
        testHurdleRace();
    }
}

class TestCase {
    int k;
    List<Integer> height;
    int expected;

    public TestCase(int k, List<Integer> height, int expected) {
        this.k = k;
        this.height = height;
        this.expected = expected;
    }
}
